go/integration_test/processor/processor_gas_test.go

```diff
@@ -0,0 +1,503 @@
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package processor
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/processor/floria"
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	op "github.com/ethereum/go-ethereum/core/vm"
+)
+
+func getGasTestScenarios() map[string]Scenario {
+	const (
+		codePrice = 3 + 3
+		excessGas = 100
+	)
+
+	allTestCases := map[string]Scenario{
+		"ValueTransferExact": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas,
+				Value:     tosca.NewValue(3),
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(97), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(3)},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas,
+			},
+		},
+		"ValueTransferSufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + excessGas,
+				Value:     tosca.NewValue(3),
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(97), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(3)},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas + excessGas/10, // 1/10th of the gas is always consumed
+			},
+		},
+		"ValueTransferInsufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas - 1,
+				Value:     tosca.NewValue(3),
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+			},
+			Receipt: tosca.Receipt{
+				Success: false,
+				GasUsed: 0,
+			},
+			OperaError: fmt.Errorf("gas too low"),
+		},
+		"SimpleCodeExact": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
+				},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + codePrice,
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
```

Comment:
why not use `PUSH0` since it should be implemented in all interpreters ?

Reply:
I would prefer the `PUSH1` version as it is here, since it would make those tests independent of the revision.